### ISSUE TITLE
Attach note to a Project (or File)

### ISSUE THREAD

**Issue by sunjam** [nextcloud/notes#598]
**Is your feature request related to a problem? Please describe.**
I write a note, which pertains to a particular file or folder on the filesystem.  I'd like to be able to attach, or associate, this note to the file / folder in some simple and direct way while writing.

**Describe the solution you'd like**
Perhaps this might become possible through #331 and Nextcloud Text implementation.  Could also involve one of the options below...

**Describe alternatives you've considered**
- Browse to the `/notes/mynote.txt` location on the hard drive and save a copy of the note in a nearby directory or something.
- Browse to the `/notes/mynote.txt` location on the hard drive and leave a comment.
- #74 is support for attachments in notes.  This approach is a bit different: to write the note and then attach it to something else on the file system.
- [System wide tagging](https://github.com/nextcloud/server/issues/13330) could expand note `tags` in the future.
- Projects functionality might eventually relate to this.

**Comment by stefan-niedermann** [comment]
I'd like to throw in another idea. What do you think about supporting Nextcloud Projects? This would not only cover your mentioned use case, but would also allow to add a note to a Deck card, Deck board, Talk chatroom and each other "object" in Nextcloud (of course only when the correspond app supports it).

It seemed to be an overkill in the past, but in the meantime it looks like a nice addition for a deeper integration with other apps and it might (partially) also solve https://github.com/nextcloud/notes/issues/389 - @sunjam and @korelstar what do you think?

Also cc @jancborchardt

**Comment by sunjam** [comment]
> What do you think about supporting Nextcloud Projects?

I actually did Project support as an option in the original post, but realize the functionality is not there yet.

> This would not only cover your mentioned use case, but would also allow to add a note to a Deck card, Deck board, Talk chatroom and each other "object" in Nextcloud (of course only when the correspond app supports it).

I didn't mention this because it adds a bit of complexity and additional integration, much like the request for system tags we've both mentioned.  There are already markdown notes available in Deck, Tasks, Bookmarks, Contacts, etc., but none of those are not accessible from the Notes app, regardless of Project support.

**Comment by stefan-niedermann** [comment]
Updare: Since Nextcloud 25 it seems that **Nextcloud Projects** are *deprecated* in favor of **Related Resources**: https://github.com/nextcloud/related_resources/

**Comment by sunjam** [comment]
Yep, we shall see if so[me similar functionality might become available through Related Resources](https://github.com/nextcloud/related_resources/issues/4).

**Comment by stefan-niedermann** [comment]
@sunjam why closing the issue though? Doesn't the Notes app have to implement anything and the Related Resources will became available magically? 😉

**Comment by sunjam** [comment]
Related resources operates on a neural network afaik, so sort of magic. Can
re-open.

**Comment by stefan-niedermann** [comment]
I assume that at least this panel needs to be integrated into the sidebar, no?

![Uploading Screenshot_20221101-090217_Fennec.png…]()
